### PR TITLE
Earth2Studio Extension Examples

### PR DESCRIPTION
**Pull request by NickGeneva**
# Earth2Studio Pull Request

## Description

Adding section of examples on extending Earth2Studio focused on just the implementation details, not the results.
I wish the diagnostic one was simpler to not include the workflow, for the future:

Also lots of updates to the install guide. Updating the model section to tabs w/ code blocks and adding some recommend environment details. Not fully tested yet

image
 

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/earth2studio/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/earth2studio/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/earth2studio/issues) is linked to this pull request.

### CODE REVIEW

File: docs/install/index.md [hunk 1]
```diff
@@ -22,40 +21,139 @@ cd earth2-inference-studio
...
+## Modulus Docker Container
+
+The recommend environment to run Earth2Studio in is the [Modulus docker container](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/modulus/containers/modulus).
+This is the way the team develops with and is the primary test bed.
+You can install Earth2Studio in a running container directly:
+
+```bash
+docker run -i -t nvcr.io/nvidia/modulus/modulus:24.04
+
+>>> pip makani[all] @ git+https://github.com/NVIDIA/modulus-makani.git@v0.1.0
```
Comment (dallasfoster) [inline docs/install/index.md]: `pip install makani[all]`?

File: docs/modules/index.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # API Reference
 
 This is the API reference for Earth-2 Inference Studio. Please refer to the full user
-guide and examples for further details on how to this package as API documentaiton
+guide and examples for further details on how to use this package as API documentaiton
 alone my not provide sufficient information.
```
Comment (dallasfoster) [inline docs/modules/index.md]: alone *may* not provide ...

File: docs/userguide/features/diagnostic.md [hunk 3]
```diff
@@ -52,33 +54,7 @@ x, coords = model(x, coords)  # Predict a single time-step
 Integrating your own diagnostic is easy, just satisfy the inferface above.
 We recommend users have a look at the custom prognostic example which will step users
 through the simple process of implementing you're own diagnostic model for your personal
```
Comment (dallasfoster) [inline docs/userguide/features/diagnostic.md]: *your* own diagnostic model ...

File: docs/userguide/features/diagnostic.md [hunk 4]
```diff
@@ -52,33 +54,7 @@ x, coords = model(x, coords)  # Predict a single time-step
...
-    :alt:
-
-  :ref:`sphx_glr_examples_01_simple_workflow.py`
-
-.. raw:: html
-
-      <div class="sphx-glr-thumbnail-title">Running Simple Inference</div>
-    </div>
-```
+needs in the {ref}`extension_examples` examples.
```
Comment (dallasfoster) [inline docs/userguide/features/diagnostic.md]: Not sure this sentence makes sense?

File: docs/userguide/features/prognostic.md [hunk 5]
```diff
@@ -71,33 +71,7 @@ for step, (x, coords) in enumerate(model_iterator):
...
-    :alt:
-
-  :ref:`sphx_glr_examples_01_simple_workflow.py`
-
-.. raw:: html
-
-      <div class="sphx-glr-thumbnail-title">Running Simple Inference</div>
-    </div>
-```
+needs in the {ref}`extension_examples` examples.
```
Comment (dallasfoster) [inline docs/userguide/features/prognostic.md:74]: As above, maybe just rephrase?

"We recommend users have a look at the custom prognostic example in {ref}`extension_examples`, which will step users through the simple process of implementing you're own prognostic model."

File: earth2studio/__init__.py [hunk 6]
```diff
@@ -14,4 +14,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-__version__ = "0.1.0a0"
+__version__ = "0.1.0"
```
Comment (dallasfoster) [inline earth2studio/__init__.py]: Not `0.2.0`?

File: docs/install/index.md (hunk 1 above)
Comment (NickGeneva) [reply docs/install/index.md]: Fixed! Thanks!

File: earth2studio/__init__.py (hunk 6 above)
Comment (NickGeneva) [reply earth2studio/__init__.py]: The change log is just place holding right now. Since initial release has no changes from a previous.

**Pull request merged**